### DISCUSSION TITLE
RBAC with Gofr

### DISCUSSION THREAD

**Discussion by dwivedisshyam** [gofr-dev/gofr#932] [Q&A] [answered]
Does Gofr supports Role-based access control? Something like gin where we can group the endpoints and use middleware to check the role required for that group?

**Comment by dwivedisshyam** [comment]
@srijan-27 ^

**Reply by aryanmehrotra** [reply]
@dwivedisshyam it is not supported, because when we are dealing with microservices. We can have a different microservice where we don't want any auth or something.
But you can write your own middleware to handle that.

But we are open- if you can tell about the use case we can discuss on that if we want to have it in GoFr🙂

**Comment by aryanmehrotra** [comment, marked as answer]
GoFr now supprots rbac through middleware checkout https://gofr.dev/docs/advanced-guide/rbac
